### ISSUE TITLE
Refactor and modularize CLI code to improve maintainability

### ISSUE THREAD

**Issue by chilanti** [appsody/appsody#29]
Items to be considered include:
* Isolating all the Docker interactions to a single module
* Isolating all the Kubernetes interactions to single module
* Revisiting the handling and retrieval of the user "home" directory
* Revisiting the "dry run" logic - evaluate whether it can be isolated rather than being dispersed around the entire code base

**Comment by kylegc** [comment]
I agree this absolutely needs to be done - soon. Making this an epic.

**Comment by EnriqueL8** [comment]
@neeraj-laad we could look at tackling this? We can use https://github.com/mibk/dupl to find all the places we have duplications and create issues for this epic.
